drivers/sensor/kx022/kx022.c

```diff
@@ -0,0 +1,881 @@
+	kx022_setting.tap_enable = (rd_data & KX022_MASK_CNTL1_TDTE) >> 2;
+	kx022_setting.motion_detect_enable = (rd_data & KX022_MASK_CNTL1_WUFE) >> 1;
+	kx022_setting.tilt_enable = rd_data & KX022_MASK_CNTL1_TPE;
+	
+	/* data output rate */
+	if(data->hw_tf->read_reg(data, KX022_REG_ODCNTL, &rd_data) < 0){
+		LOG_DBG("KX022 : Failed to read ODCNTL");
+		return -EIO;
+	}
+	kx022_setting.ODR= rd_data & KX022_MASK_ODCNTL_OSA;
+
+	/* interrupt 1 setting */
+	if(data->hw_tf->read_reg(data, KX022_REG_INC1, &rd_data) < 0){
+		LOG_DBG("KX022 : Failed to read INC1");
+		return -EIO;
+	}
+	kx022_setting.pin_1.enable = (rd_data & KX022_MASK_INC5_IEN2) >> 5;
+	kx022_setting.pin_1.polarity = (rd_data & KX022_MASK_INC5_IEA2) >> 4;
+
+	/* interrupt 1  event */
+	if(data->hw_tf->read_reg(data, KX022_REG_INC4, &rd_data) < 0){
+		LOG_DBG("KX022 : Failed to read INC4");
+		return -EIO;
+	}
+	kx022_setting.pin_1.tap = (rd_data & KX022_MASK_INC4_TDTI1) >> 2; 
+	kx022_setting.pin_1.motion_detect = (rd_data & KX022_MASK_INC4_WUFI1) >> 1; 
+	kx022_setting.pin_1.tilt = rd_data & KX022_MASK_INC4_TPI1; 
+
+	/* interrupt 2 setting */
+	if(data->hw_tf->read_reg(data, KX022_REG_INC5, &rd_data) < 0){
+		LOG_DBG("KX022 : Failed to read INC5");
+		return -EIO;
+	}
+	kx022_setting.pin_2.enable = (rd_data & KX022_MASK_INC5_IEN2) >> 5;
+	kx022_setting.pin_2.polarity = (rd_data & KX022_MASK_INC5_IEA2) >> 4;
+
+	/* interrupt 2 event */
+	if(data->hw_tf->read_reg(data, KX022_REG_INC6, &rd_data) < 0){
+		LOG_DBG("KX022 : Failed to read INC6");
+		return -EIO;
+	}
+	kx022_setting.pin_2.tap = (rd_data & KX022_MASK_INC4_TDTI1) >> 2; 
+	kx022_setting.pin_2.motion_detect = (rd_data & KX022_MASK_INC4_WUFI1) >> 1; 
+	kx022_setting.pin_2.tilt = rd_data & KX022_MASK_INC4_TPI1; 
+
+	/* sampling rate */
+	if(data->hw_tf->read_reg(data, KX022_REG_LP_CNTL, &rd_data) < 0){
+		LOG_DBG("KX022 : Failed to read LP_CNTL");
+		return -EIO;
+	}
+	kx022_setting.sampling_rate = (rd_data & KX022_MASK_LP_CNTL_AVC) >> 4;
+
+	kx022_data_get(val, 2, kx022_setting.operating_mode, kx022_setting.resolution);
+	kx022_data_get(val + 1, 2, kx022_setting.acceleration_range, kx022_setting.ODR);
+	kx022_data_get(val + 2, 2, kx022_setting.tap_enable, kx022_setting.motion_detect_enable);
+	kx022_data_get(val + 3, 2, kx022_setting.tilt_enable, kx022_setting.sampling_rate);
+
+	kx022_data_get(val + 4, 2, kx022_setting.pin_1.enable, kx022_setting.pin_1.polarity);
+	kx022_data_get(val + 5, 2, kx022_setting.pin_1.tap, kx022_setting.pin_1.motion_detect);
+	kx022_data_get(val + 6, 2, kx022_setting.pin_1.tilt, 0);
+
+	kx022_data_get(val + 7, 2, kx022_setting.pin_2.enable, kx022_setting.pin_2.polarity);
+	kx022_data_get(val + 8, 2, kx022_setting.pin_2.tap, kx022_setting.pin_2.motion_detect);
+	kx022_data_get(val + 9, 2, kx022_setting.pin_2.tilt, 0);
+
+	return 0;
+}
+
+static int kx022_accel_motion_get(const struct device *dev, struct sensor_value *val)
+{
+	struct kx022_data *data = dev->data;
```

Comment:
You will be able to get per instance configuration once you move that struct inside `struct kx022_data`, i.e.:

```c
struct kx022_configuration *config = data->configurations;

/* Then access the member like so */
config->motion_detect.xn_direction
```